front/src/modules/ui/modal/components/Modal.tsx

```diff
@@ -95,18 +101,44 @@ const modalVariants = {
 export function Modal({
   isOpen = false,
   children,
-  onOutsideClick,
+  closeModal,
+  hotkeyScope = ModalHotkeyScope.Default,
   ...restProps
 }: ModalProps) {
   const modalRef = useRef<HTMLDivElement>(null);
 
   useListenClickOutside({
     refs: [modalRef],
-    callback: () => onOutsideClick?.(),
+    callback: () => closeModal?.(),
     mode: ClickOutsideMode.absolute,
   });
 
-  if (!isOpen) {
+  const {
+    goBackToPreviousHotkeyScope,
+    setHotkeyScopeAndMemorizePreviousScope,
+  } = usePreviousHotkeyScope();
+
+  useScopedHotkeys(
+    [Key.Escape],
+    () => {
+      closeModal?.();
+    },
+    hotkeyScope,
+    [closeModal],
+  );
+
+  useScopedHotkeys(
+    [Key.Enter],
```

Comment:
Pressing Enter closes and opens the modal which is not desired, I'm removing the Enter hotkey but it might seem a bit hacky